sched/task/task_tls_alloc.c

```diff
@@ -0,0 +1,113 @@
+/****************************************************************************
+ * sched/task/task_tls_alloc.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <sys/types.h>
+#include <unistd.h>
+#include <sched.h>
+#include <errno.h>
+
+#include <nuttx/tls.h>
+#include <nuttx/semaphore.h>
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+#if CONFIG_TLS_TASK_NELEM > 0
+
+static tls_ndxset_t g_tlsset = 0;
```

Comment:
We either need to add
```
#if CONFIG_TLS_TASK_NELEM > 0
#  if CONFIG_TLS_TASK_NELEM > 64
#    error Too many task TLS elements
#  elif CONFIG_TLS_TASK_NELEM > 32
     typedef uint64_t tls_task_ndxset_t;
#  elif CONFIG_TLS_TASK_NELEM > 16
     typedef uint32_t tls_task_ndxset_t;
#  elif CONFIG_TLS_TASK_NELEM > 8
     typedef uint16_t tls_task_ndxset_t;
#  else
     typedef uint8_t tls_task_ndxset_t;
# endif
```
and use `tls_task_ndxset_t` here or just use `uint64_t` for simplicity.

Reply:
Thanks